### PR TITLE
Add FilterResult interface and Status.filtered optional prop

### PR DESCRIPTION
**Pull request by ayoayco**
This PR will
1. add the optional `filtered` prop to Status https://docs.joinmastodon.org/entities/Status/#filtered
2. add a new interface FilterResult https://docs.joinmastodon.org/entities/FilterResult/

### CODE REVIEW

**Comment by neet** [comment]
@ayoayco CI seems to be failing due to Prettier warning. Can you `yarn run test:eslint --fix`?

**Comment by ayoayco** [comment]
Sorry missed that. Fixed!

File: src/entities/filter-result.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ * Represents a filter whose keywords matched a given status.
+ * @see https://docs.joinmastodon.org/entities/filter-result/
+ */
+export interface FilterResult {
+  /** The filter that was matched. */
+  filter: Filter;
+  /** The keyword within the filter that was matched. */
+  keywordMatches: string[] | null;
+  /** The status ID within the filter that was matched. */
+  status_matches: string[] | null;
```
Comment (neet) [inline src/entities/filter-result.ts:13]: It should be `statusMatches`

**Review by neet — CHANGES_REQUESTED** [review]

File: src/entities/filter-result.ts (hunk 1 above)
Comment (ayoayco) [reply src/entities/filter-result.ts]: Fixed!

**Comment by ayoayco** [comment]
If I need to add tests, we can keep this open and I will do it maybe tomorrow. :)

**Comment by neet** [comment]
I also added `filterAction` in #742

**Pull request merged**